v5.12.2 (#5223)
mui/mui-x: CHANGELOG.md

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## v5.12.2

_Jun 16, 2022_

> We'd like to offer a big thanks to the 7 contributors who made this release possible. Here are some highlights ✨:

Reviewer: Should we highlight the community contributors?

- 🚀 Performance improvement for dynamic row height (#5135) @m4theushw
- 📚 Documentation improvements
- 🐞 Bugfixes

> ### `@mui/x-data-grid@v5.12.2` / `@mui/x-data-grid-pro@v5.12.2` / `@mui/x-data-grid-premium@v5.12.2`

Author: This is the first time. I did not updated the pickers version

> ### `@mui/x-data-grid@v5.12.2` / `@mui/x-data-grid-pro@v5.12.2` / `@mui/x-data-grid-premium@v5.12.2`

Reviewer 2: Ok, then we won't release them right? I'm asking because I thought they are automatically released but I guess if the version doesn't change then they won't be released.

> ### `@mui/x-data-grid@v5.12.2` / `@mui/x-data-grid-pro@v5.12.2` / `@mui/x-data-grid-premium@v5.12.2`

Reviewer: Since the date pickers have no synchronized version with the grid, it's okay to not release it. From v6 on, after we sync them, I think we should bump the version anyway, but that's a discussion beyond this issue, so we can reserve some time to think about how we're going to approach this situation in the future.

> ### `@mui/x-data-grid@v5.12.2` / `@mui/x-data-grid-pro@v5.12.2` / `@mui/x-data-grid-premium@v5.12.2`

Reviewer 3: The current strategy for synced packages is to not release if there is no change

Sometime you can see that we don't release the license package
And the next time we will bump it of two versions to re-establish the sync.

> ### `@mui/x-data-grid@v5.12.2` / `@mui/x-data-grid-pro@v5.12.2` / `@mui/x-data-grid-premium@v5.12.2`

Reviewer 4: The opposite may happen someday: no changes for the data grid but few changes for the date pickers. In this case, even the tags sequence will be broken if we don't bump them. For v6, data grid and date pickers will have to be bumped together.

#### Changes

- [DataGrid] Fix for cosmetic bug on column filter badge (#5170) @simbahandiane
- [DataGrid] Hide vertical scrollbar if `autoHeight` is enabled (#5164) @m4theushw
- [DataGrid] Use exponential search to render non-measured rows (#5135) @m4theushw
- [DataGridPro] Fix label of the detail panel toggle column (#5191) @m4theushw
- [DataGridPro] Refresh detail panel caches when props change (#5110) @m4theushw

### Docs

- [docs] Add example with custom checkbox column (#5161) @flaviendelangle
- [docs] Batch small changes (#5177) @oliviertassinari
- [docs] Fix demo currency format (#5034) @oliviertassinari
- [docs] Fix redundant headers (#5104) @oliviertassinari
- [docs] Fix some capitalization to match the guidelines (#5105) @oliviertassinari
- [docs] Improve the `getRowId` doc section (#5156) @flaviendelangle
- [docs] Use redirection in the code (#5114) @oliviertassinari

> - [docs] Using data grid with date pickers (#5053) @cherniavskii

Reviewer 4: ```suggestion
- [docs] Add demo of how to use the data grid with date pickers (#5053) @cherniavskii
```

> 

Reviewer: ```suggestion
- [docs] Instruction to deal with invalid license message (#5074) @joserodolfofreitas
- [docs] Fix outdated license description to match the EULA (#5219) @joserodolfofreitas

```

### Core

- [core] Improve inline code rendering within the details tag (#5166) @Harmouch101
- [core] Remove unused props from plugin typing (#5185) @flaviendelangle
- [core] Use the scrollbar size from `useGridDimensions` on `DataGridProColumnHeaders` (#5201) @flaviendelangle
- [core] Fix GridColTypeDef type (#5167) @cherniavskii

> - [core] Fix GridColTypeDef usage in demo (#5197) @cherniavskii

Reviewer 2: ```suggestion
- [core] Fix `GridColTypeDef` usage in demo (#5197) @cherniavskii
```
